Makefile

```diff
@@ -161,8 +163,12 @@ push-operator: operator-manager
 
 push: push-handler push-operator
 
-test/unit: $(GO)
-	NODE_NAME=node01 $(GINKGO) $(unit_test_args) $(WHAT)
+.PHONY: $(addprefix test-unit-submodules-,$(WHAT))
+$(addprefix test-unit-submodules-,$(WHAT)):
+	# hande unit test for submodule
+	(cd $(subst test-unit-submodules-,,$@) ; NODE_NAME=node01 $(GINKGO) $(unit_test_args) `$(GO) list -f {{.Dir}} ./...`)
+
+test/unit: $(GO) $(addprefix test-unit-submodules-,$(WHAT))
```

Comment:
Relaxing a little the WHAT filter we can do 

```Makefile
test/unit/api: $(GO)                                                            
    cd api && $(GINKGO) $(unit_test_args) ./...                                 
                                                                                
test/unit: test/unit/api                                                        
    NODE_NAME=node01 $(GINKGO) $(unit_test_args) $(WHAT) 
```

Reply:
The reason i did it this way was because of `pkg/names`, i had to make it a submodule as well because api importing from it, so if it's not a submodule it will create a cyclic dependency. 
Running `NODE_NAME=node01 $(GINKGO) $(unit_test_args) $(WHAT)` on pkg will fail because it will not be able to run the unit tests of `pkg/names`

Reply:
I forgot to add pkg/names to `WHAT`, added that as well. unit tests passed locally.

Reply:
Let's move `pkg/names` to `api/names`, having a submodule only for `pkg/names` does not make any sense then we can do the Makefile change:

```Makefile
test/unit/api: $(GO)                                                            
    cd api && $(GINKGO) $(unit_test_args) ./...                                 
                                                                                
test/unit: test/unit/api                                                        
    NODE_NAME=node01 $(GINKGO) $(unit_test_args) $(WHAT) 
```

Reply:
I was thinking that it's not related to the `api` package but sure no problem

Reply:
fixed